pkg/proxy/keepalive/timeout_darwin.go

```diff
@@ -0,0 +1,34 @@
+//go:build darwin
+
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keepalive
+
+import (
+	"syscall"
+	"time"
+
+	"github.com/pingcap/TiProxy/lib/util/errors"
+)
+
+func setTimeout(syscn syscall.RawConn, d time.Duration) error {
+	var serr error
+	return errors.Collect(ErrTimeout, serr, syscn.Control(func(fd uintptr) {
+		serr = syscall.SetsockoptInt(int(fd), syscall.IPPROTO_TCP, syscall.TCP_RXT_CONNDROPTIME, int(d.Seconds()))
+		if serr != nil {
+			return
+		}
```

Comment:
Why need this?
BTW, should TCP_CONNECTIONTIMEOUT be also set?

Reply:
I am not sure about if `TCP_CONNECTIONTIMEOUT` is the correct parameter for apple and BSD. But `TCP_RXT_CONNDROPTIME` is more similar to `TCP_USER_TIMEOUT`.

Reply:
I saw that for BSD it's TCP_KEEPINIT. https://gitlab.isc.org/isc-projects/bind9/-/issues/2334

Reply:
I saw that TCP_CONNECTIONTIMEOUT is for creating new connections. We already have dial timeout, so we don't need that.